provider/provider.go

```diff
@@ -287,19 +287,18 @@ func (p *dockerNativeProvider) Diff(ctx context.Context, req *rpc.DiffRequest) (
 func diffUpdates(updates map[resource.PropertyKey]resource.ValueDiff) map[string]*rpc.PropertyDiff {
 	updateDiff := map[string]*rpc.PropertyDiff{}
 	for key, valueDiff := range updates {
-		if string(key) != "registry" {
-			updateDiff[string(key)] = &rpc.PropertyDiff{
-				Kind: rpc.PropertyDiff_UPDATE,
-			}
-		} else {
+		skipUpdate := false
```

Comment:
Nit: we have lots of negatives here. If we flip `skipUpdate` to `update`, we get code that IMO is easier to read:
```suggestion
		update := true
```

Then we on lines L296:
```patch
-			_, hasServer := valueDiff.Object.Updates["server"]
-			skipUpdate = !hasServer
+			_, update = valueDiff.Object.Updates["server"]
```

And L299
```patch
-		if !skipUpdate {
+ 		if update {
```

Reply:
Great suggestion, done!